drop-boxes/register-fasta-dropbox/register-fasta.py

```diff
@@ -138,11 +135,10 @@ def process(transaction):
 
         for f in os.listdir(incomingPath):
             if f.endswith('origlabfilename'):
-                origName = open(os.path.join(incomingPath,f), 'r')
-                secondaryName = origName.readline().strip().split('_')[0]
-                origName.close()
-                sa.setPropertyValue('Q_SECONDARY_NAME', secondaryName)
-                os.remove(os.path.realpath(os.path.join(incomingPath,f)))
+                with open(os.path.join(incomingPath,f), 'r') as origName:
+                  secondaryName = origName.readline().strip().split('_')[0]
+                  sa.setPropertyValue('Q_SECONDARY_NAME', secondaryName)
+                  os.remove(os.path.realpath(os.path.join(incomingPath,f)))
```

Comment:
not sure you can remove it while it's open, i would put this (and maybe the line before) outside the "with" block

Reply:
Yes